paddle/fluid/framework/details/multi_devices_graph_builder.cc

```diff
@@ -187,15 +188,53 @@ std::unique_ptr<SSAGraph> MultiDevSSAGraphBuilder::Build(
   };
 
   bool is_forwarding = true;
+  int rpc_op_device_id = 0;
+  auto schedule_rpc_op = [&]() -> void {
+    rpc_op_device_id++;
+    if (rpc_op_device_id >= static_cast<int>(places_.size())) {
+      rpc_op_device_id = 0;
+    }
+  };
+
   for (auto *op : program.Block(0).AllOps()) {
     if (boost::get<int>(
             op->GetAttr(OpProtoAndCheckerMaker::OpRoleAttrName())) ==
         static_cast<int>(OpRole::kRPC)) {
       // append rpc op if program is distributed trainer main program.
       // always use the first device
-      CreateRPCOp(&result, *op);
+      if (op->Type() == "send_vars") {
+        auto got = remote_vars_devices_.find(op->InputArgumentNames()[0]);
+        if (got == remote_vars_devices_.end()) {
+          schedule_rpc_op();
+        } else {
```

Comment:
When is ```else``` triggered?  I guess you want to round-robin the send devices?

Not quite remember. In Reduce mode, is gradients calculated in 1 device instead of all devices? Should the send device match that device?

Reply:
`DistributedTranspiler` would split a parameter into some parameter blocks, but this is not applicable to each parameter if it's small enough, we didn't split it. So the op pipeline is such as:
```bash
(split_byref)->send->recv->(concat)
```

For the true block of the if statement: there is no `split_byref` operator before `send`, so we need to schedule `send` to the right place.
For the false block, `split_byref` has already been schedule to a device, and `send` should be scheduled to the same device with `split_byref`